src/main/java/io/aiven/kafka/connect/gcs/RecordGrouper.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Aiven Kafka GCS Connector
+ * Copyright (c) 2019 Aiven Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.aiven.kafka.connect.gcs;
+
+import org.apache.kafka.connect.sink.SinkRecord;
+
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * The interface for classes that associates {@link SinkRecord}s with files by some criteria.
+ */
+interface RecordGrouper {
+    /**
+     * Associate the record with the appropriate file.
+     */
+    void put(SinkRecord record);
+
+    /**
+     * Clear all records.
+     */
+    void clear();
+
+    /**
+     * Get all file names.
+     */
+    Collection<String> fileNames();
```

Comment:
Returning some container that has both the file names and the records for each file (maybe just `Map<String, List<Record>>`) would imo be cleaner than returning file names first and then explicitly asking records per file. This interface requires the grouper to be implemented in a specific way. (E.g. it might be reasonable not to do any file name splitting dynamically when new records are added but this interface kind of requires it.)

Reply:
Good point. I don't have any particular reason making them separate. Changed.